### ISSUE TITLE
Separate "flake8-polifyll" to extra requirement

### ISSUE THREAD

**Issue by tribals** [rubik/radon#189]
When installing `radon`, it requires `flake8-polyfill`, which, in turn, requires `flake8`.

IMO, this should be an option provided though extra requirement (something like `radon[flake8]`) since not everyone want `flake8` installed when using just `radon` alone.

**Comment by rubik** [comment]
This is a fair point, but then I don't know how Radon would declare the `flake8.extention` entry point in a conditional way (see setup.py). If you have a workaround, I can implement it.

**Comment by tribals** [comment]
You can't declare entry point in conditional way, as depended on extra requirement. But you don't need to.

Instead, you can [declare extra requirements for entry points](https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-extras-optional-features-with-their-own-dependencies), too. Example:

> Extras can be used by a project’s entry points to specify dynamic dependencies. For example, if Project A includes a “rst2pdf” script, it might declare it like this, so that the “PDF” requirements are only resolved if the “rst2pdf” script is run:

```python

setup(
    name="Project-A",
    ...
    entry_points={
        "console_scripts": [
            "rst2pdf = project_a.tools.pdfgen [PDF]",
            "rst2html = project_a.tools.htmlgen",
            # more script entry points ...
        ],
    }
)
```

Entry point declared in such a way still will be discoverable (e.g. `pkg_resources.get_entry_info()`). But it is not a problem since the only one who can import it is extra requirement (`flake8` in this case), and it will not be installed unless `pip` directly instructed to do so. And when it is installed, this is still not a problem since both entry point and extra requirement already in place. So, `flake8` will be able to import entry point without any issues.

**Comment by rubik** [comment]
@tribals Thanks for the PR Anthony! I didn't know that entry points could contain extra requirements. This change will be included in the next release.

**Comment by tribals** [comment]
I didn't know too. Until opened this issue. :grin:

**Comment by cblegare** [comment]
If `flake8_polyfill` is optional, it should be hidden a conditional import

here, for instance

https://github.com/rubik/radon/blob/master/radon/complexity.py#L7

**Comment by tribals** [comment]
Good point. I should have been check for this before PR.

**Comment by tribals** [comment]
Take a look, please: #200. What do you think?

**Comment by rubik** [comment]
@cblegare @tribals My bad, I should have caught it in the previous PR. I merged the new PR and released radon 4.3.2 which contains the fix. I have verified that it works by installing radon in a clean new environment.

**Comment by cblegare** [comment]
Thanks to both of you :)
